test/Microsoft.ML.Benchmarks/StochasticDualCoordinateAscentClassifierBench.cs

```diff
@@ -31,31 +30,43 @@ public class StochasticDualCoordinateAscentClassifierBench : WithExtraMetrics
             PetalWidth = 5.1f,
         };
 
-        private Legacy.PredictionModel<IrisData, IrisPrediction> _trainedModel;
+        private TransformerChain<MulticlassPredictionTransformer<MulticlassLogisticRegressionModelParameters>> _trainedModel;
+        private PredictionEngine<IrisData, IrisPrediction> _predictionEngine;
         private IrisData[][] _batches;
-        private ClassificationMetrics _metrics;
+        private MultiClassClassifierMetrics _metrics;
 
         protected override IEnumerable<Metric> GetMetrics()
         {
             if (_metrics != null)
                 yield return new Metric(
-                    nameof(ClassificationMetrics.AccuracyMacro),
+                    nameof(MultiClassClassifierMetrics.AccuracyMacro),
                     _metrics.AccuracyMacro.ToString("0.##", CultureInfo.InvariantCulture));
         }
 
         [Benchmark]
-        public Legacy.PredictionModel<IrisData, IrisPrediction> TrainIris() => Train(_dataPath);
+        public TransformerChain<MulticlassPredictionTransformer<MulticlassLogisticRegressionModelParameters>> TrainIris() => Train(_dataPath);
 
-        private Legacy.PredictionModel<IrisData, IrisPrediction> Train(string dataPath)
+        private TransformerChain<MulticlassPredictionTransformer<MulticlassLogisticRegressionModelParameters>> Train(string dataPath)
         {
-            var pipeline = new Legacy.LearningPipeline();
+            var env = new MLContext(seed: 1, conc: 1);
+            var reader = new TextLoader(env,
```

Comment:
Can you use a single mlContext for both training and predicting? Would require defining it as a global variable, and using it in both training and prediction code.